### PR TITLE
boost python lib name: use the same policy for Apple, NetBSD & ArchLinux

### PR DESCRIPTION
**Pull request by nim65s**
Hi,

This PR follows #168, and similarly adds a `CHECK_NETBSD` to find the correct name for the boost python component.

### CODE REVIEW

**Comment by nim65s** [comment]
Hi,

ArchLinux has the same behavior,  and provides boost_python / boost_python3 / boost_python27 / boost_python37 but no boost_python2, so I updated the PR accordingly

File: boost.cmake [hunk 1]
```diff
@@ -83,13 +85,13 @@ MACRO(SEARCH_FOR_BOOST)
       ENDIF(NOT PYTHONLIBS_FOUND)
       IF(${PYTHON_VERSION_MAJOR} EQUAL 3) 
         LIST(REMOVE_AT BOOST_COMPONENTS_ ${PYTHON_IN_BOOST_COMPONENTS})
-        IF(APPLE)
+        IF(APPLE OR NETBSD_FOUND)
```
Comment (gergondet) [inline boost.cmake]: So, shouldn't that also apply to `ARCHLINUX_FOUND`?

Comment (nim65s) [reply boost.cmake]: I missed it, but on the other hand, ArchLinux currently provides boost v1.68 so the ELSE(Boost_VERSION VERSION_GREATER 106699) can be ignored

Comment (gergondet) [reply boost.cmake]: Ok, no problem then :) (sorry, I didn't check the bigger scope of this part of the script)

**Pull request merged**